dialog/dialog/relational-algebra.md

```diff
@@ -0,0 +1,415 @@
+]
+```
+
+## 3.9 Equijoin
+
+An equijoin is a binary operator which is parameterized over a [propositional formula](#4-propositional-formula), where the [propositional formula](#4-propositional-formula) only makes use of equality. It computes the set of all combinations of tuples in its input relations for which the given attribute names are equal. The relations MUST have disjoint attributes, and the tuples in the resulting set are flattened.
+
+This operation MAY be implemented using [theta join](#38-theta-join), however it is RECOMMENDED that equijoins are specialized, for performance reasons.
+
+```
+categories = [
+    (category_id: 1, category_name: "Electronics"),
+    (category_id: 2, category_name: "Books")
+]
+
+products = [
+    (product_id: 1, product_category_id: 1, product_name: "Tele-Multiplay 6000"),
+    (product_id: 2, product_category_id: 1, product_name: "Xerox 8"),
+    (product_id: 3, product_category_id: 2, product_name: "Foucault's Pendulum"),
+    (product_id: 4, product_category_id: 2, product_name: "100 Years of Solitude")
+]
+
+equijoin(category_id = product_category_id, products) => [
+    (category_id: 1, category_name: "Electronics", product_id: 1, product_category_id: 1, product_name: "Tele-Multiplay 6000"),
+    (category_id: 1, category_name: "Electronics", product_id: 2, product_category_id: 1, product_name: "Xerox 8"),
+    (category_id: 2, category_name: "Books", product_id: 3, product_category_id: 2, product_name: "Foucault's Pendulum"),
+    (category_id: 2, category_name: "Books", product_id: 4, product_category_id: 2, product_name: "100 Years of Solitude")
+]
+```
+
+## 3.10 Semijoin
+
+A semijoin is a binary operator which returns all tuples in its first input relation such that there is a tuple in its second relation for which the common attributes of both tuples are equal.
+
+This operation is equivalent to composing a [projection](#31-projection) again the first relation's attributes with a [natural join](#37-natural-join), however implementations are RECOMMENDED to implement semijoins as a specialization of these operations, for performance reasons.
+
+For example:
+
+```
+users = [
+    (user_id: 1, user_name: "Alice"),
+    (user_id: 2, user_name: "Bob"),
+    (user_id: 3, user_name: "Mallory")
+]
+
+admins = [
+    (user_id: 1),
+    (user_id: 2),
+]
+
+semijoin(users, admins) [
+    (user_id: 1, user_name: "Alice"),
+    (user_id: 2, user_name: "Bob"),
+]
+```
+
+## 3.11 Antijoin
+
+An antijoin is a binary operator which returns all tuples in its first input relation such that there is no tuple in its second relation for which the common attributes of both tuples are equal.
+
+This operation is equivalent to taking the [difference](#35-difference) of the first relation and the [semijoin](#310-semijoin) of both, however implementations MAY implement antijoins as a specialization of these operations, for performance reasons.
```

Comment:
> implementations MAY implement antijoins as a specialization

SHOULD this be a RECOMENDED?

What is happening to me :sweat_smile:

Reply:
Yup, I'll change that. Thanks!